picoshare/file_lifetime.go

```diff
@@ -19,6 +19,12 @@ type FileLifetime struct {
 // not expire, effectively.
 var FileLifetimeInfinite = NewFileLifetimeInYears(100)
 
+func NewFileLifetimeFromDuration(duration time.Duration) FileLifetime {
+	return FileLifetime{
```

Comment:
Can we move this check in `parse.GuestFileLifetime` to here:

```go
	if dur < (time.Hour * 1) {
		return picoshare.FileLifetime{}, ErrFileLifeTimeTooShort
	}
```

That way, we have consistent enforcement to prevent the caller from creating a `FileLifetime` shorter than 1 hour.

Reply:
Done, but to be able to pass the correct error properly I had to do this:
``` golang
if duration < (time.Hour * 24) {
	return FileLifetime{}, fmt.Errorf("file lifetime must be at least 1 day")
}
```
Also I noticed that we don't even support file lifetimes shorter than 1 day so I adjusted the duration accordingly. (Not sure why I had it previously set to 1 hour)